app/controllers/handoff/RegistrationConfirmationController.scala

```diff
@@ -52,11 +53,11 @@ class RegistrationConfirmationController @Inject()(val authConnector: PlayAuthCo
         registered {
           regid =>
             handBackService.decryptConfirmationHandback(requestData) flatMap {
-              case Success(s) => handBackService.storeConfirmationHandOff(s, regid).flatMap {
+              case payload@Success(s) => handBackService.storeConfirmationHandOff(s, regid).flatMap {
                 case _ if handBackService.payloadHasForwardLinkAndNoPaymentRefs(s) => getPaymentHandoffResult(externalID)
-                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show))
-                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show))
-                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage))
+                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(payload.get.language)))
+                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(payload.get.language)))
+                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage).withLang(Lang(payload.get.language)))
```

Comment:
```suggestion
                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(s.language)))
                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(s.language)))
                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage).withLang(Lang(s.language)))
```

Reply:
You could rename the inner extraction on the pattern matching to be `payload` if you wanted it to be consistent and more understandable than what was there originally as `s` :)